tools/props-bot/contribution-collector.js

```diff
@@ -0,0 +1,274 @@
+const fetch = require('node-fetch');
+
+const contributors = {
+	userData: [],
+};
+const contributorTypes = [
+	'committers',
+	'reviewers',
+	'commenters',
+	'reporters',
+];
+
+module.exports = async ( { github, context } ) => {
+	for ( const type of contributorTypes ) {
+		contributors[ type ] = new Set();
+	}
+
+	/*
+	 * Fetch the following data for the pull request:
+	 * - Commits with author details.
+	 * - Reviews with author logins.
+	 * - Comments with author logins.
+	 * - Linked issues with author logins.
+	 * - Comments on linked issues with author logins.
+	 */
+	const contributorData = await github.graphql(
+		`query($owner:String!, $name:String!, $prNumber:Int!) {
+			repository(owner:$owner, name:$name) {
+				pullRequest(number:$prNumber) {
+					commits(first: 100) {
+						nodes {
+							commit {
+								author {
+									user {
+										databaseId
+										login
+										name
+										email
+									}
+									name
+									email
+								}
+							}
+						}
+					}
+					reviews(first: 100) {
+						nodes {
+							author {
+								login
+							}
+						}
+					}
+					comments(first: 100) {
+						nodes {
+							author {
+								login
+							}
+						}
+					}
+					closingIssuesReferences(first:100){
+						nodes {
+							author {
+								login
+							}
+							comments(first:100) {
+								nodes {
+									author {
+										login
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}`,
+		{
+			owner: 'WordPress',
+			name: 'gutenberg',
+			prNumber: 54416,
```

Comment:
It might be easier to run the testing on this PR, to see whether the comment updates whenever someone adds a comment, commit, etc.

Reply:
I switched it back to using `context` values. I had changed it just to test a PR with more activity on it from more contributors.